### PR TITLE
Refactor and add tests

### LINKED ISSUE #4: Automatic testing of all current supported formats
Would be nice to have automated tests for XBM and netpbm formats.

### PR DESCRIPTION
**Pull request by martinkauppinen**
Resolves #4. At least partially.

Basic tests added for all supported formats and variants. Minor bugfixes and clippy warnings squashed.

### CODE REVIEW

File: src/parser/netpbm/pbm.rs [hunk 1]
```diff
@@ -1,38 +1,43 @@
...
+                let pix = self.data[(i * self.header.width + j) as usize];
 
-            if pix > 0 {
-                print!("\x1b[48;2;{r};{g};{b}m  ");
-            } else {
-                print!("\x1b[0m  ");
+                if pix > 0 {
+                    pxbm_write!(writer, "{color}")?;
+                } else {
+                    pxbm_write!(writer, "{}", Color::new(None))?;
```
Comment (kjeller) [inline src/parser/netpbm/pbm.rs]: Everything with Color struct is better than this mess `print!("\x1b[0m  ");`...I like how readable the write macro & Color combination is (even though I don't fully understand how to read macro definitions :)).

With that said, I think I would prefer something like `Color::transparent()` or something along those lines to make it more explicit in what it is supposed to do. Color::transparent() could even use the `Color::new(None)` handling.
What do you think?

File: src/parser/netpbm/mod.rs [hunk 2]
```diff
@@ -32,40 +31,232 @@ pub struct NetpbmHeader {
...
+impl<Writer: Write> Parser<Writer> for Netpbm {
+    fn print(&self, color: Color, writer: &mut Writer) -> Result<()> {
         match self.header.filetype {
-            MagicNumber::P1 => pbm::print_netpbm_p1(self, r, g, b),
-            MagicNumber::P2 => pgm::print_netpgm(self),
-            MagicNumber::P3 => ppm::print_netppm(self),
-            MagicNumber::P4 => pbm::print_netpbm_p4(self, r, g, b),
-            MagicNumber::P5 => pgm::print_netpgm(self),
-            MagicNumber::P6 => ppm::print_netppm(self),
+            MagicNumber::P1 => self.print_netpbm_p1(color, writer),
```
Comment (kjeller) [inline src/parser/netpbm/mod.rs:67]: I thought it looked nice with ascending order. Grouping by matched calls makes sense as well

**Comment by kjeller** [comment]
Forgot to submit review... comments should be visible now

File: src/parser/netpbm/pbm.rs (hunk 1 above)
Comment (martinkauppinen) [reply src/parser/netpbm/pbm.rs]: `Color::transparent()` is much nicer than having `Option` arguments to the normal constructor. Implemented.

File: src/parser/netpbm/mod.rs (hunk 2 above)
Comment (martinkauppinen) [reply src/parser/netpbm/mod.rs:67]: I believe this change was suggested by the linter. I don't remember, it was months ago at this point 😅

**Pull request merged**